### PR TITLE
Fix: context menus on Firefox

### PR DESCRIPTION
**Pull request by bershanskiy**
Firefox does not support `contextMenus` as an optional permission. It must be among required `permissions` in manifest.

[MDN docs](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/optional_permissions).

### CODE REVIEW

**Comment by alexanderby** [comment]
Will the menu show up by default?

**Comment by bershanskiy** [comment]
No, menus will be disabled by default. We technically could make menus enabled by default, but (1) I don't like when extensions do that and clutter the UI and (2) this would not be consistent with Chrome, potentially causing user confusion.

**Review by Gusted — CHANGES_REQUESTED** [review]
We should add a note in the ui just for firefox users that the permission won't be requested as it can't be a optional permissions as some will definitely open issues about it :)

**Comment by bershanskiy** [comment]
Chrome (no changes)

image

Firefox (extra message only for "OFF" state)
image

File: src/ui/popup/settings-page/context-menus.tsx [hunk 1]
```diff
@@ -29,7 +33,7 @@ export default function ContextMenusGroup(props: ViewProps) {
             label="Use context menus"
             description={props.data.settings.enableContextMenus ?
                 'Context menu integration is enabled' :
-                'Context menu integration is disabled'}
+                `Context menu integration is disabled${isFirefox ? '\nNote: conext menus permission is not used.' : ''}`}
```
Comment (Gusted) [inline src/ui/popup/settings-page/context-menus.tsx:36]:
```suggestion
                `Context menu integration is disabled${isFirefox ? '\nNote: context menus permission is not used.' : ''}`}
```

**Review by Gusted — CHANGES_REQUESTED** [review]

File: src/ui/popup/settings-page/context-menus.tsx (hunk 1 above)
Comment (bershanskiy) [reply src/ui/popup/settings-page/context-menus.tsx]: Good catch!

**Review by Gusted — APPROVED** [review]

**Pull request merged**